gate-web/src/main/groovy/com/netflix/spinnaker/gate/interceptors/RequestSheddingInterceptor.java

```diff
@@ -0,0 +1,161 @@
+ * interceptor will elect to drop all low priority requests.
+ */
+public class RequestSheddingInterceptor extends HandlerInterceptorAdapter {
+
+  final static String PRIORITY_HEADER = "X-Spinnaker-Priority";
+  final static String LOW_PRIORITY = "low";
+
+  final static String ENABLED_KEY = "requestShedding";
+  final static String CHANCE_KEY = "requestShedding.chance";
+  final static String PATHS_KEY = "requestShedding.paths";
+
+  private final static Logger log = LoggerFactory.getLogger(RequestSheddingInterceptor.class);
+  private final static Splitter splitter = Splitter.on(",").omitEmptyStrings().trimResults();
+  private final static Random random = new Random();
+
+  private final DynamicConfigService configService;
+  private final Registry registry;
+
+  private final Id requestsId;
+
+  public RequestSheddingInterceptor(DynamicConfigService configService, Registry registry) {
+    this.configService = configService;
+    this.registry = registry;
+
+    this.requestsId = registry.createId("requestShedding.requests");
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request,
+                           HttpServletResponse response,
+                           Object handler) throws Exception {
+    if (configService.isEnabled(ENABLED_KEY, false) && isDroppable(request)) {
+      if (shouldDropRequestWithChance()) {
+        log.warn("Dropping low priority request: {}", request.getRequestURI());
+        registry.counter(requestsId.withTag("action", "dropped")).increment();
+
+        response.setDateHeader("Retry-After", Instant.now().plus(Duration.ofMinutes(1)).toEpochMilli());
+        throw new LowPriorityRequestRejected();
+      }
+      log.debug("Dice roll prevented low priority request shedding: {}", request.getRequestURI());
+      registry.counter(requestsId.withTag("action", "allowed")).increment();
+    }
+
+    return super.preHandle(request, response, handler);
+  }
+
+  private boolean isDroppable(HttpServletRequest request) {
+    String uri = request.getRequestURI();
+
+    // We want our error handler to service the errors correctly
+    if ("/error".equals(uri)) {
+      return false;
+    }
+
+    return LOW_PRIORITY.equals(request.getHeader(PRIORITY_HEADER)) &&
+      hasPathMatch(uri);
+  }
+
+  private boolean shouldDropRequestWithChance() {
+    Integer chance = configService.getConfig(Integer.class, CHANCE_KEY, 100);
+    if (chance < 0 || chance > 100) {
+      log.warn("Request shedding drop chance is out of bounds: {}, assuming 100", chance);
+      chance = 100;
+    }
+    return random.nextInt(100) <= chance;
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  private boolean hasPathMatch(String requestedPath) {
+    String pathMatchers = configService.getConfig(String.class, PATHS_KEY, "");
+    if (Strings.isNullOrEmpty(pathMatchers)) {
+      return true;
+    }
+
+    List<String> paths = splitter.splitToList(pathMatchers);
+    if (paths.isEmpty()) {
+      return true;
```

Comment:
that means empty paths pattern matches everything... have you considered matching nothing instead?

Reply:
I considered matching nothing instead, but since we'll only have a small subset of requests that are actually getting the header with a `low` value, I think we'll want to be able to wholesale drop low priority requests, rather than first having to inventory what requests are low priority, then adding the list of requests.

Clients are informing us that the request is low priority, so we should have the ability to drop them all if we need to.

Reply:
isn't that what `/*` is for?

Reply:
Sure. I can get behind that ... Suppose better safe than sorry. Will make the change.